projects/02-meme-generator-api/solution/lib/memeapi.rb

```diff
@@ -0,0 +1,37 @@
+require 'sinatra/base'
+require './lib/download'
+require './lib/request_image'
+require 'pry'
+require './lib/validator'
+
+class MemeApi < Sinatra::Application
+
+  class << self
+    def run
+      run!
+    end
+  end
+
+  post '/memes' do
+    #req = RequestImage.new(request.body.read)
+    #halt req.status, req.message if req.status == 400
+    #image_name = req.create_image
+    json_body = JSON.parse(request.body.read)
+    validation = Validator.validate_image(json_body)
+    unless validation
+      halt 400, validation.message
+    end
+    image_path = Download.download_image(json_body['meme']['image_url'])
+    image_name = ImageCreator.create_meme(image_path, json_body['meme']['text'])
+    redirect "/memes/#{image_name}", 303
+    rescue Download::Error => e
+      halt 404, e.message
+  end
+
+
+  get '/memes/:file' do
+    path = Pathname(__dir__).join("../images/#{params[:file]}")
```

Comment:
The implementation here is way too low level IMO. I suggest you try to implement the repository for reading and writing to a file and use it here.
That way it is more isolated and we can link together the piece that reads with the piece that writes, and if need to change, we would be able to do it in a single place.